### PR TITLE
Speech command recognition - Quadratic Self-ONN

### PR DESCRIPTION
**Pull request by jovaroo**
Adds an implementation of a Quadratic Self-ONN based learner, which can be used for keyword spotting or other limited-vocabulary speech recognition tasks.

### CODE REVIEW

**Comment by passalis** [comment]
@jovaroo  testing is now available again. So, when you are ready, you can use the "test sources" label to run the style checks and "test tools" to run the tool tests. 
 
Please also let us known when this PR is ready for review.

**Comment by jovaroo** [comment]
@passalis The tests now run successfully and the tool is ready to be reviewed!

File: src/opendr/perception/speech_recognition/quadraticselfonn/algorithm/model.py
Comment (passalis) [inline src/opendr/perception/speech_recognition/quadraticselfonn/algorithm/model.py:1]: This probably need to be updated to 2020-2021.

File: src/opendr/perception/speech_recognition/quadraticselfonn/quadraticselfonn_learner.py
Comment (passalis) [inline src/opendr/perception/speech_recognition/quadraticselfonn/quadraticselfonn_learner.py:220]: There is no need to include the external path/folder name into the model_paths field of the json file, since this is already known (since we know where the model folder is).

**Review by passalis — COMMENTED** [review]
Thanks for this PR. I have left a few minor comments regarding this implementation.

**Comment by jovaroo** [comment]
Addressed both of your concerns, @passalis! The test pipeline now fails, but it seems like error is related to another tool "ERROR: test_eval (tools.perception.pose_estimation.lightweight_open_pose.test_lightweight_open_pose.TestLightweightOpenPoseLearner)".

The exact error message is "TypeError: 'numpy.float64' object cannot be interpreted as an integer", which makes it sound an awful lot like a version difference (most likely numpy) in the pipeline. The reason why this began to happen on commit 231632917437754c7cff383a214113f55bab779e after only changing the copyright years and one unrelated line of code is beyond me. Any suggestions on how to proceed with the issue?

**Comment by passalis** [comment]
Hi @jovaroo, the reason for getting these errors now is that probably the tests didn't run in the previous test runs (I opened issue #103 in order to investigate this). Regarding the issue with the numpy compatibility, I patched the version of pycocotools used in pose estimation and I opened the PR #104 that uses the patched version. So, hopefully this will resolve the issue that you are reporting.

**Comment by jovaroo** [comment]
Thanks, @passalis, now the tests work again as intended. Or at the very least least in an order favourable for this branch :)

**Review by passalis — APPROVED** [review]
Thanks!

**Review by stefaniapedrazzi — CHANGES_REQUESTED** [review]
I just have a couple of comments about the documentation.

By the way I tried to run your example and it doesn't work.
I get this error for the first code snippet:
```
Traceback (most recent call last):
  File "test_a.py", line 15, in <module>
    learner = QuadraticSelfOnnLearner(output_clases_n=10, iters=10, expansion_order=3)
TypeError: __init__() got an unexpected keyword argument 'output_clases_n'
```

And this in the second one (not fully tested because the training failed):
```
Traceback (most recent call last):
  File "test_a.py", line 6, in <module>
    learner.load(os.path.join(".", "example", "directory", "path", "model"))
NameError: name 'os' is not defined
```

**Comment by stefaniapedrazzi** [comment]
Please also synchronize with master so that the tests are run on the latest version.

File: docs/reference/quadratic-selfonn.md
Comment (jovaroo) [reply docs/reference/quadratic-selfonn.md:53]: Fixed code to match this

**Comment by jovaroo** [comment]
I applied the suggestions made @stefaniapedrazzi, fixed the examples and tested them on my console. However, the tests seem to fail on other tools again in the CI pipeline. A similar error can be seen on other branches, too, for example at https://github.com/tasostefas/opendr_internal/runs/3067747810. Ping @passalis FYI.

File: docs/reference/quadratic-selfonn.md
Comment (stefaniapedrazzi) [inline docs/reference/quadratic-selfonn.md:213]: Maybe it would be good to also explain where does this file come from.
Because running this snippet as-is it throws the errors:
```
RuntimeError: Error opening 'command.wav': System error.
...
FileNotFoundError: [Errno 2] No such file or directory: 'command.wav'
```

**Review by stefaniapedrazzi — DISMISSED** [review]
Thank you!

I confirm the test errors are due to the `object_tracking_2d` tests (#114) and not related with this branch. I disabled the `object_tracking_2d` tests locally and everything works correctly.

I could now also run correctly the first code example.
I could not test the second one due to the missing file

File: docs/reference/quadratic-selfonn.md
Comment (jovaroo) [reply docs/reference/quadratic-selfonn.md:214]: I added a note for the reader to clarify the source of the file in the example

**Review by stefaniapedrazzi — APPROVED** [review]
Thank you

**Comment by stefaniapedrazzi** [comment]
This PR is waiting for a second review.
@passalis do you want to review this PR?

**Review by passalis — APPROVED** [review]
All my concerns have been addressed. Thank you!

**Pull request merged**
